spec/associations/has_many_nested_spec.rb

```diff
@@ -0,0 +1,323 @@
+          )
+        }
+      end
+
+      it 'generates flattened structures to use as paths for entries' do
+        vcr('association/nested_without_root_parentless_higher_include') {
+          nested = Nested.query.params('sys.id' => '5CNbxcca5OwgI68eSuWso2', include: 3).load.first
+          child_1 = nested.children.first
+          child_2 = child_1.children.first
+          expect(nested.all_child_paths_by(:name)).not_to be_empty
+          expect(nested.all_child_paths_by(:id)).to eq [[child_1.id, child_2.id]]
+        }
+      end
+
+      it 'generates a path for for entries that match a value' do
+        vcr('association/nested_without_root_parentless_higher_include') {
+          nested = Nested.query.params('sys.id' => '5CNbxcca5OwgI68eSuWso2', include: 3).load.first
+          child_1 = nested.children.first
+          child_2 = child_1.children.first
+          expect(nested.find_child_path_by(:name, child_2.name)).not_to be_empty
+          expect(nested.find_child_path_by(:name, child_2.name)).to eq [[child_1.name, child_2.name]]
+
+          expect(nested.find_child_path_by(:name, 'foo')).to be_empty
+          expect(nested.find_child_path_by(:name, 'foo')).to eq []
+        }
+      end
+    end
+
+    context 'has parent and child' do
+      it 'parent exists' do
+        vcr('association/nested_without_root_middle') {
+          nested = Nested.find('3gpECMvSLSOC8wgkQ6GQ2q')
+          expect(nested.parent).to be_a Nested
+          expect(nested.parent.name).to eq 'Nested 1'
+        }
+      end
+
+      it 'has children' do
+        vcr('association/nested_without_root_middle') {
+          nested = Nested.find('3gpECMvSLSOC8wgkQ6GQ2q')
+
+          expect(nested.children?).to be_truthy
+          expect(nested.children).to be_a ::Array
+          expect(nested.children.size).to eq 1
+          expect(nested.children.first).to be_a Nested
+          expect(nested.children.first.name).to eq 'Nested 3'
+        }
+      end
+
+      it 'is not the root' do
+        vcr('association/nested_without_root_middle_parent') {
+          nested = Nested.find('3gpECMvSLSOC8wgkQ6GQ2q')
+          expect(nested.root?).to be_falsey
+          expect(nested.root).to eq nested.parent
+        }
+      end
+
+      it 'has ancestors' do
+        vcr('association/nested_without_root_middle_parent') {
+          nested = Nested.find('3gpECMvSLSOC8wgkQ6GQ2q')
+          expect(nested.ancestors).not_to be_empty
+          expect(nested.ancestors.size).to eq 1
+          expect(nested.ancestors.first.name).to eq 'Nested 1'
+        }
+      end
+
+      it 'find ancestors will return an enumerator of ancestors - if no block given' do
+        vcr('association/nested_without_root_middle_parent') {
+          nested = Nested.find('3gpECMvSLSOC8wgkQ6GQ2q')
+          expect(nested.find_ancestors.to_a.map(&:name)).to eq ['Nested 1']
+        }
+      end
+
+      it 'find ancestos execute the block for each ancestor - if block given' do
```

Comment:
`ancestos` -> `asbestos`

I also think there is a word missing: `find ancestors` **to** `execute`?